tests/upgrade-tests/utils.py

```diff
@@ -41,10 +41,11 @@ def dump_model_info(model):
 def assert_no_unit_errors(model):
     for unit in model.units.values():
         workload_status = unit.data['workload-status']['current']
-        juju_status = unit.data['juju-status']['current']
         assert workload_status != 'error'
-        assert juju_status != 'failed'
-        assert juju_status != 'lost'
+        if 'juju-status' in unit.data:
```

Comment:
In juju 2+, 'juju-status' has become 'agent-status'. You can use `unit.agent_status` here. And `unit.workload_status` above. See https://github.com/juju/python-libjuju/blob/master/juju/unit.py#L13 and https://github.com/juju/python-libjuju/blob/master/juju/unit.py#L34.